app/common/state/tabContentState/partitionState.js

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// State helpers
+const tabUIState = require('../tabUIState')
+const frameStateUtil = require('../../../../js/state/frameStateUtil')
+
+// Constants
+const {tabs} = require('../../../../js/constants/config')
+
+module.exports.isPartitionTab = (state, frameKey) => {
+  const frame = frameStateUtil.getFrameByKey(state, frameKey)
+
+  if (frame == null) {
+    if (process.env.NODE_ENV !== 'test') {
+      console.error('Unable to find frame for isPartitionTab method')
+    }
+    return false
+  }
+
+  return !!frame.get('partitionNumber')
+}
+
+module.exports.getPartitionNumber = (state, frameKey) => {
+  const frame = frameStateUtil.getFrameByKey(state, frameKey)
+
+  if (frame == null) {
+    if (process.env.NODE_ENV !== 'test') {
+      console.error('Unable to find frame for getPartitionNumber method')
+    }
+    return 0
+  }
+  const partitionNumber = frame.get('partitionNumber')
+  if (typeof partitionNumber === 'string') {
```

Comment:
like @NejcZdovc mentioned... I believe the logic of actually parsing the number out exists already in `frameStateUtil.getPartitionNumber()`

Reply:
not really. That method does not accept numbers or number-like arguments. So a partition passing down `1` or `'1'` will return `0`. I did a fiddle to demonstrate: https://jsfiddle.net/cezaraugusto/okrgjtok/